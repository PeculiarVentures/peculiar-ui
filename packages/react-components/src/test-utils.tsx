import React from 'react';
import {
  render,
  screen,
  fireEvent,
  waitFor,
  renderHook,
  act,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from './styles/theme_provider';

const renderWithWrapper = (
  ui: React.ReactElement,
) => (
  render(
    ui,
    {
      wrapper: ThemeProvider,
    },
  )
);

export {
  userEvent,
  fireEvent,
  renderWithWrapper,
  screen,
  waitFor,
  renderHook,
  act,
};
