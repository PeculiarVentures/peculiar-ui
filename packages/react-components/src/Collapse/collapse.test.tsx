import React from 'react';
import { renderWithWrapper as render } from '../test-utils';
import { Collapse } from './index';

describe('<Collapse />', () => {
  it('should render with default styles', () => {
    const { asFragment } = render(<Collapse>Inside</Collapse>);

    expect(asFragment()).toMatchSnapshot();
  });
});
