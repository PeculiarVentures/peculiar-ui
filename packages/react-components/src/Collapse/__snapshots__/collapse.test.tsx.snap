// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Collapse /> should render with default styles 1`] = `
<DocumentFragment>
  .emotion-0 {
  overflow: hidden;
  -webkit-transition: height 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: height 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  height: 0;
  visibility: hidden;
}

<div
    class="emotion-0"
  >
    <div>
      Inside
    </div>
  </div>
</DocumentFragment>
`;
